packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -0,0 +1,444 @@
+        (_) => Future(() => standardAndroidPushMessage),
+      );
+      await plugin.configure(
+        authProviderRepo: authProviderRepo,
+        config: config,
+      );
+      expect(log, <Matcher>[
+        isMethodCall('listen', arguments: null),
+        isMethodCall('cancel', arguments: null),
+        isMethodCall('listen', arguments: null),
+      ]);
+      verify(
+        mockServiceProviderClient.recordNotificationEvent(
+          eventType: anyNamed('eventType'),
+          notification: anyNamed('notification'),
+        ),
+      ).called(1);
+    });
+
+    test('should throw exception when configuring if there is no appId present',
+        () async {
+      final config0 = AmplifyConfig.fromJson(
+        jsonDecode(noPushAppIdAmplifyConfig) as Map<String, Object?>,
+      );
+      expect(
+        () async => plugin.configure(
+          authProviderRepo: authProviderRepo,
+          config: config0,
+        ),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+    });
+
+    test('should fail configure when registering device is unsuccessfull',
+        () async {
+      log.clear();
+      testWidgetsFlutterBinding.defaultBinaryMessenger.setMockMethodCallHandler(
+        MethodChannel(tokenReceivedEventChannel.name),
+        throwErrorHandler,
+      );
+      expect(
+        () async => plugin.configure(
+          authProviderRepo: authProviderRepo,
+          config: config,
+        ),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+    });
+
+    test('should throw PushNotificationException if not configured', () async {
+      expect(
+        () async => plugin.onTokenReceived,
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.onNotificationReceivedInForeground,
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.onNotificationOpened,
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.launchNotification,
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.identifyUser(
+          userId: 'userId',
+          userProfile: AnalyticsUserProfile(),
+        ),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.requestPermissions(),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.getPermissionStatus(),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+
+      expect(
+        () async => plugin.setBadgeCount(42),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
+      expect(
+        () async => plugin.getBadgeCount(),
+        throwsA(const TypeMatcher<PushNotificationException>()),
+      );
```

Comment:
These two assertion actually caught the exception thrown for `!_iosCheck` I believe, which made the test inaccurate...

Reply:
The ios check only logs an error and doesn't throw